### ISSUE TITLE
Project's license: redeclare as PSF-2.0

### ISSUE THREAD

**Issue by befeleme** [zopefoundation/roman#15]
## FEATURE REQUEST

_TL;DR: Python 2.1.1 is not a valid SPDX identifier, would it be possible to redeclare the project as PSF-2.0?_

Hello,
Fedora Linux is going through the process of declaring the licenses according to the SPDX standard. 
I was pointed here from docutils where we've came across the `roman`'s license: Python 2.1.1, which is not an existing SPDX identifier. Having asked Fedora Legal for an opinion, it was pointed out that the original declaration by Mark Pilgrim was this:

> "The example programs in this book are free software; you can redistribute and/or modify them under the terms of the Python license as published by the Python Software Foundation. A copy of the license is included in Appendix H, Python license."

And suggestion by the Legal team member:

> I would suggest they replace the Python 2.1.1 license text with some sort of instantiation of the PSF-2.0 license text, and possibly attempt to get in touch with Mark Pilgrim to confirm that this is appropriate.

Would it be possible?

[docutils issue](https://sourceforge.net/p/docutils/feature-requests/95/)
[Fedora Legal issue](https://gitlab.com/fedora/legal/fedora-license-data/-/issues/216#note_1387584572)

**Comment by dataflake** [comment]
I have a suggestion since you're the party desiring this change: You get in touch with Mark Pilgrim and ask him if that's OK and ask him to either add a comment to that effect here or email zope-dev@zope.dev (or you forward  his answer to you to zope-dev@zope.dev). I'll be happy to make the change then.

**Comment by icemac** [comment]
@pilgrim-brave Thank you for your permit.

@befeleme I created a PR in #16. Is this the change you requested?
There is also https://github.com/zopefoundation/roman/blob/master/LICENSE.txt. Can this file be kept a is or does it also need changes?

**Comment by dataflake** [comment]
Yes, I believe the license text file must be changed, otherwise the declared license no longer matches the shipped license file. The license file contains the "Python 2.0 license" (see https://www.python.org/download/releases/2.0/license/), which is not the same as PSF-2.0, which is the "Python Software Foundation License 2.0".

I found the license text for the PSF-2.0 license at https://wiki.python.org/moin/PythonSoftwareFoundationLicenseV2Easy and here's a link that shows how to modify the PSF for use in projects that are not owned by the PSF: https://wiki.python.org/moin/PythonSoftwareFoundationLicenseFaq#How_do_I_use_the_PSF_License.3F

**Comment by befeleme** [comment]
> @befeleme I created a PR in #16. Is this the change you requested? There is also https://github.com/zopefoundation/roman/blob/master/LICENSE.txt. Can this file be kept a is or does it also need changes?

Please also change the license file to achieve consistency. Thank you!

**Comment by icemac** [comment]
@dataflake Thank you for digging up the licence and providing a howto to alter it. I did it in #16 but the text feels wrong. It seems no longer to be PSF-2.0 but more like a completely new licence. Could we alternatively use the actual [Zope Public License](https://github.com/zopefoundation/Zope/blob/master/LICENSE.txt) (ZPL-2.1) instead?

**Comment by dataflake** [comment]
This is a question for @befeleme.

To me it looks like the original license declaration by @pilgrim-brave was already impossible to implement - the Python 2.0 license as well as the PSF-2.0 represent a license contract between the PSF and the end user. But this software package was never owned by the PSF so that made no sense back then and it makes no sense now.

The PSF license howto says you should edit the license text and not use it as-is. I do not know if it's even valid to call such a changed license text "PSF-2.0" anymore. I'm not a lawyer but would guess no.

If I had a choice I would ask Mark if it's OK to use the Zope Public License version 2.1 instead. The package is already in the GitHub "zopefoundation" organization and it's one of a handful (out of 300+ packages) that does not use the ZPL. The ZPL is also OSI certified: https://opensource.org/license/zpl-2-1/.

@pilgrim-brave would you agree to change the license to the ZPL version 2.1? I would also add a dedicated copyright notice file that explicitly names "Mark Pilgrim and contributors" as copyright holder and add a copyright note at the top of all source code files similar to what we have in all other projects, something like this:

```
##############################################################################
#
# Copyright (c) 2001 Mark Pilgrim and Contributors.
# All Rights Reserved.
#
# This software is subject to the provisions of the Zope Public License,
# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
# FOR A PARTICULAR PURPOSE.
#
##############################################################################
```

**Comment by befeleme** [comment]
> This is a question for @befeleme.

I have no objections to any desired license change that's agreed by all parties (not counting self as a party). My initial query is about the inconsistency between the license declaration from the original source and the license text provided. As an RPM package maintainer, I'll welcome any consistent open-source outcome. Zope Public License version 2.1 is [allowed in Fedora](https://docs.fedoraproject.org/en-US/legal/allowed-licenses/).

**Comment by dataflake** [comment]
@befeleme Take a look at #16 and let me know what you think.

**Comment by dataflake** [comment]
I just released version 4.1 with the new license.
